packages/SwingSet/src/kernel/state/kernelKeeper.js

```diff
@@ -1202,6 +1324,41 @@ export default function makeKernelKeeper(kernelStorage, kernelSlog) {
     return makeUpgradeID(nextID);
   }
 
+  function addTerminatedVat(vatID) {
+    if (!terminatedVats.includes(vatID)) {
+      terminatedVats.push(vatID);
+      kvStore.set(`vats.terminated`, JSON.stringify(terminatedVats));
+    }
+  }
+
+  function getFirstTerminatedVat() {
```

Comment:
Should we be using "vatID" rather than "vat"?
```suggestion
  function getFirstTerminatedVatID() {
```
but also, is this even carrying its own weight as a distinct function rather than `nextCleanupTerminatedVatAction` checking the `terminatedVats` array directly?

Reply:
I think it's clear from context that `get*Vat` is returning a VatID.. there's no single JS object which "is" the vat (there's a VatKeeper, and a Manager for the worker, but the "vat" is too abstract to be a single returnable object), so I think we can allow "Vat" in method names to stand in for the most commonly referenced Vat-related thing, which would be its ID.

You're correct, this function isn't really carrying its weight, but 1: it references an in-RAM cache (backed by the kvStore), which makes it important to hide the details from callers, 2: it's exported, but 3: nothing uses it, I think I exported it for the sake of a test that I didn't write, but 4: I should write that test.